ra/1.1/resource-agent-api.md

```diff
@@ -5,118 +5,129 @@ AT https://github.com/ClusterLabs/OCF-spec/ TO DISCUSS CHANGES**
+to user queries.
 
-### "Resource types"
+### Resource Agent
 
-A resource type represents a set of resources which share a common set of
-instance parameters and a common set of actions which can be performed on
-resource of the given type.
+A _resource agent_ (RA) is a software application implementing the RA API for a
+particular resource type. An RA allows a resource manager to perform specific
+mangement tasks for resource instances.
 
-The resource type name is chosen by the provider of the RA.
+### Resource Agent Provider
 
+A _resource agent provider_ is an entity supplying one or more resource agents
+for installation on cluster hosts. Each provider should have a unique name
+suitable for use as a file system directory name.
 
-### "Resource agent"
+A resource agent provider beginning with a leading dot (.) is a hint to RMs and
+other tools that the resource agent provider should be omitted from lists
+provided in response to user queries.
 
-A RA provides the actions ("member functions") for a given type of
-resources; by providing the RA with the instance parameters, it is used
-to control a specific resource.
+A provider may choose to supply multiple, separate collections of resource
+agents. In this case, each collection should have a unique name, and _provider_
+may refer either to the entity as a whole, or to an individual collection.
 
-They are usually implemented as shell scripts, but the API described here does
-not require this.
+Currently, there is no central registry for provider names. Providers should
+choose names that do not appear to be already in use for publicly available
+resource agents.
 
-Although this is somewhat similar to LSB init scripts, there are some
-differences explained below. 
+Each provider also chooses the resource type names used for the resource agents
+it provides. These do not need to be unique across providers.
 
+### Resource Name
 
-### "Instance parameters"
+A _resource name_ is a unique identifier chosen by the cluster administrator
+to identify a particular resource instance.
 
-Instance parameters are the attributes which describe a given resource
-instance. It is recommended that the implementor minimize the set of
-instance parameters.
+### Resource Parameters
 
-The meta data allows the RA to flag one or more instance parameters as
-`unique`. This is a hint to the RM or higher level configuration tools
-that the combination of these parameters must be unique to the given
-resource type.
+_Resource parameters_, also known as _instance parameters_, are attributes
+describing a particular resource instance. Each parameter has a name and a
+value, which must satisfy the requirements of POSIX environment variable names
+and values.
 
-An instance parameter has a given name and value. They are both case
-sensitive and must satisfy the requirements of POSIX environment
-name/value combinations.
+The resource agent defines the names, meaning, and allowed values of parameters
```

Comment:
optional: s/meaning/meanings/ ? Either works.

Reply:
Updated "corrections" commit with this fix (and carried it over to 1.1 in updated "bring in proposed changes" commit)